### ISSUE TITLE
SoilProfileCollection horizon setter

### ISSUE THREAD

**Issue by brownag** [ncss-tech/aqp#82]
If this issue is encompassed elsewhere (#6 is close) then I am sorry.

I recognize a lot of these improvements to the SPC have been tagged as "aqp 2.0" -- but in the meantime there is probably a need for a simple implementation of this functionality in 1.x

I want a similar left-join like syntax for horizon level attributes as we have for site.

e.g

```
new.data <- some_function_to_fetch_ancillary_horizon_data()
horizons(spc) <- new.data
```

This would make relation to e.g. pedon horizon lab results child tables easy. Generally, incorporating multiple attributes at once at the horizon level.

Currently, `site()`-setter allows for left join on any attribute name containing the unique ID. While I like the ease-of-application of that approach, I wonder if that approach is too broad for `horizons()`. Thoughts?

Also, I have wondered about allowing joins based on names attribute from named vectors and lists, as well as the conventional data frame, essentially making use of identifying information if it is present and uniquely matches the horizon ID.

I suppose the lists would ultimately have to contain a compatible `data.frame`-like structure. That might be more effort than its worth. At a minimum, ensuring that named attributes don't get crammed into the horizon table out of order seems essential.

**Comment by dylanbeaudette** [comment]
This is a valid concern and I see no reason that it should wait for "2.0". We are a ways from getting there anyway.

Some thoughts:
   * keep it simple: joining `data.frames` is fairly robust and they are simple to make
   * I think that it should be up to the user to ensure that the *correct* ID is present in both tables
   * warnings / errors should be issued when the join results in more rows than the original
   * parity with `site<-` and `horizons<-` is important

**Comment by brownag** [comment]
I agree on all of those items. 

I guess I was alluding to two different issues.

Perhaps more than _explicitly supporting_ the named vectors and lists, it might be good just provide some output if you try to set a single attribute with a named vector that is not matching the order of horizons and encourage the user to use a `data.frame` for an explicit join. 

I guess I am thinking it is pretty easily to generate named vectors and lists from various apply based functions etc. It is fine if your variable is calculated from a consistent set of existing horizon level attributes (which are inherently ordered correctly). BUT... if those ID's get out of order -- with the length match the number of rows in horizon table -- there is nothing stopping the user from putting them in out of order. 

I do agree the data.frame format is simple and should be the desired focal datatype for horizon setter like this.

**Comment by brownag** [comment]
Closed with 1ffdabb
